types/tablet_v2/wlr_tablet_v2_tool.c

```diff
@@ -792,18 +792,6 @@ static void implicit_tool_up(struct wlr_tablet_tool_v2_grab *grab) {
 	check_and_release_implicit_grab(grab);
 }
 
-/* Only send the motion event, when we are over the surface for now */
-static void implicit_tool_motion(
-	struct wlr_tablet_tool_v2_grab *grab, double x, double y) {
-	struct implicit_grab_state *state = grab->data;
-	if (state->focused != state->original) {
```

Comment:
Hmm. But using the default handler would send the event to the wrong surface then?

Reply:
Before this change pressure also used the default handler, and surfaces _did_ get the pressure events when implicitly grabbed (just not the motion ones, and there were some more issues addressed in swaywm/sway#5303).

Reply:
Hmm, I still don't get why this change is required. If events are sent to the right surface, then `state->focused == state->original` should be true, right?

Reply:
I think `state->focused` is misleadingly named, it's not actually the surface that is focused (as far as sway knows), it's actually "which surface *would* have focus if we weren't in an implicit grab", i.e. the surface currently under the pen.

https://github.com/swaywm/wlroots/blob/973cb19ab643649bdb32a2aa5e09bbe727205fb6/types/tablet_v2/wlr_tablet_v2_tool.c#L767-L779

Reply:
Ah, indeed! Thanks for clearing this up.